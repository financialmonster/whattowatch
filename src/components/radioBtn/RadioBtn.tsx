import React, { FC, ChangeEvent, memo } from 'react';

type TRadioBtnProps = {
    value: number;
    groupValue: string;
    disabled: boolean;
    field: {
        onChange: (e: ChangeEvent<any>) => void;
        onBlur: () => void;
    }
}

export const RadioBtn: FC<TRadioBtnProps> = memo(({ value, groupValue, disabled, field: {onChange, onBlur}}) => (
    <>
        <input className="rating__input" id={`star-${value}`} type="radio" name="rating" value={ value }
            checked={ value === +groupValue } onChange={onChange} onBlur={onBlur} disabled={ disabled }
        />
        <label className="rating__label" htmlFor={`star-${value}`}>
            {`Rating-${value}`}
        </label>
    </>
));
