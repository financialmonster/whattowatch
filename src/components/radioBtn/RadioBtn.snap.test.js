import React from 'react';
import renderer from 'react-test-renderer';

import { RadioBtn } from './RadioBtn';

it(`RadioBtn: correctly rendered`, () => {
    const tree = renderer.create(
        <RadioBtn value={5} groupValue="test" field={{onBlur: () => {}, onChange: () => {}}} disabled />
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
