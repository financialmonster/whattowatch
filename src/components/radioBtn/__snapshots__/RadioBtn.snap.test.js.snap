// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioBtn: correctly rendered 1`] = `
Array [
  <input
    checked={false}
    className="rating__input"
    disabled={true}
    id="star-5"
    name="rating"
    onBlur={[Function]}
    onChange={[Function]}
    type="radio"
    value={5}
  />,
  <label
    className="rating__label"
    htmlFor="star-5"
  >
    Rating-5
  </label>,
]
`;
